### PR TITLE
Support horizontal scrolling in ScrollableViewport

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

File: sky/packages/sky/lib/widgets/basic.dart [hunk 1]
```diff
@@ -268,17 +268,23 @@ class Baseline extends OneChildRenderObjectWrapper {
...
-    : super(key: key, child: child);
+  Viewport({
+    Key key,
+    this.scrollOffset: Offset.zero,
+    this.scrollDirection: ViewportScrollDirection.vertical,
+    Widget child
+  }) : super(key: key, child: child);
 
-  final double offset;
+  final Offset scrollOffset;
```
Comment (HansMuller) [inline sky/packages/sky/lib/widgets/basic.dart]: The distinction between Offset scrollOffset and double scrollOffset is confusing.  Can we (eventually?) use the Offset version everywhere?

File: sky/packages/sky/lib/widgets/scrollable.dart [hunk 2]
```diff
@@ -188,12 +195,12 @@ abstract class Scrollable extends StatefulComponent {
   }
 
   EventDisposition _handleScrollUpdate(sky.GestureEvent event) {
-    scrollBy(direction == ScrollDirection.horizontal ? event.dx : -event.dy);
+    scrollBy(scrollDirection == ViewportScrollDirection.horizontal ? event.dx : -event.dy);
```
Comment (HansMuller) [inline sky/packages/sky/lib/widgets/scrollable.dart]: What if scrollDirection is both?  

Maybe scrollBy() and scrollTo() should have an Offset valued parameter.

File: sky/packages/sky/lib/widgets/scrollable.dart [hunk 3]
```diff
@@ -370,7 +382,7 @@ abstract class FixedHeightScrollable extends Scrollable {
     return new SizeObserver(
       callback: _handleSizeChanged,
       child: new Viewport(
-        offset: offsetY,
+        scrollOffset: new Offset(0.0, offsetY),
```
Comment (HansMuller) [inline sky/packages/sky/lib/widgets/scrollable.dart]: I guess Fixed and VariableHeightScrollable will eventually need to support the vertical and horizontal ViewportScrollDirections.

**Comment by HansMuller** [comment]
LGTM. I think more work will be needed to really support scrollDirection: everywhere that it's appropriate.

File: sky/packages/sky/lib/widgets/basic.dart (hunk 1 above)
Comment (abarth) [inline sky/packages/sky/lib/widgets/basic.dart]: Yeah, we really should be using the Offset version.

File: sky/packages/sky/lib/widgets/scrollable.dart (hunk 2 above)
Comment (abarth) [inline sky/packages/sky/lib/widgets/scrollable.dart]: There's an assert that we don't get "bot" as a value, but we should support it.

File: sky/packages/sky/lib/widgets/scrollable.dart (hunk 3 above)
Comment (abarth) [inline sky/packages/sky/lib/widgets/scrollable.dart]: Yeah..

File: sky/packages/sky/lib/widgets/basic.dart (hunk 1 above)
Comment (abarth) [inline sky/packages/sky/lib/widgets/basic.dart]: This was too big a change for this CL.  I'll do it in another CL.

**Pull request merged**